include/cache.h

```diff
@@ -13,45 +13,72 @@
 extern "C" {
 #endif
 
-#define _sys_cache_flush_sig(x) void (x)(vaddr_t virt, size_t size)
-
-#if defined(CONFIG_CACHE_FLUSHING)
-
-#if defined(CONFIG_ARCH_CACHE_FLUSH_DETECT)
-	typedef _sys_cache_flush_sig(_sys_cache_flush_t);
-	extern _sys_cache_flush_t *sys_cache_flush;
-#else
-	extern _sys_cache_flush_sig(sys_cache_flush);
+/**
+ *
+ * @brief Flush d-cache lines to main memory
+ *
+ * No alignment is required for either addr or size, but since
+ * sys_cache_flush() iterates on the d-cache lines, a d-cache line alignment for
+ * both is optimal.
+ *
+ * The d-cache line size is specified either via the CONFIG_CACHE_LINE_SIZE
+ * kconfig option or it is detected at runtime.
+ *
+ * @param addr the pointer to start the multi-line flush
+ * @param size the number of bytes that are to be flushed
+ *
+ * @return N/A
+ */
+static inline void sys_dcache_flush(void *addr, size_t size)
+{
+#ifdef CONFIG_CACHE_FLUSHING
+	arch_dcache_flush(addr, size);
 #endif
+}
 
-#else
-
-/*
- * Provide NOP APIs for systems that do not have caches.
+/**
+ *
+ * @brief Invalidate d-cache lines
  *
- * An example is the Cortex-M chips. However, the functions are provided so
- * that code that need to manipulate caches can be written in an
- * architecture-agnostic manner. The functions do nothing. The cache line size
- * value is always 0.
+ * No alignment is required for either addr or size, but since
+ * sys_cache_invd() iterates on the d-cache lines, a d-cache line alignment for
```

Comment:
comment(s) need to updated in accordance with API name.

Reply:
Updated the comments and rebased. Thanks.